### PR TITLE
add benchmarking for mochiweb

### PR DESCRIPTION
**Pull request by robertkowalski**
Add a benchmark-script, and add `make bench` which creates an app
for the benchmark. Additionally add a new route to the sample app
to have a benchmark that is not dependent on the file system.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,3 +21,9 @@ clean:
 
 app:
 	@$(REBAR) -r create template=mochiwebapp dest=$(DEST) appid=$(PROJECT)
+
+bench: app
+	@echo "Writing $(DEST)/$(PROJECT).sh"
+	@cp support/bench.sh $(DEST)/bench.sh
+	@perl -p -i -e "s/-s reloader/-s reloader\n    -detached/g" $(DEST)/start-dev.sh
+	@perl -p -i -e "s/-s reloader/-s reloader \\\/g" $(DEST)/start-dev.sh
```
Comment (etrepum) [inline Makefile]: would be nice to do this without a perl dependency

Comment (robertkowalski) [inline Makefile]: good point. after thinking a while i think running erlang detached is not necessary at all - i use a dedicated machine for it (which i would really recommend for benchmarking to anyone else)

**Comment by robertkowalski** [comment]
@etrepum happy to rebase - just want to make further reviewing for both PRs not harder.

File: Makefile [hunk 2]
```diff
@@ -21,3 +21,7 @@ clean:
 
 app:
 	@$(REBAR) -r create template=mochiwebapp dest=$(DEST) appid=$(PROJECT)
+
+bench: app
```
Comment (ddosia) [inline Makefile]: Perhaps little bit late, but would it be more *Makefile*-ish to do:

```make
.PHONY: bench
bench: app $(DEST)/$(PROJECT).sh

$(DEST)/$(PROJECT).sh:
	@cp support/bench.sh $(DEST)/bench.sh
``` 
One other question, why does it says "Writing $(DEST)/**$(PROJECT)**.sh" but actually creating "$(DEST)/**bench**.sh"?

Comment (etrepum) [inline Makefile]: Not really relevant, I removed the Makefile changes altogether and just added bench.sh to the application skeleton. It didn't really make much sense where it was.

**Pull request merged**